# Set up a computer for Computational Biology

* [R, RStudio, and Bioconductor](r_bioc.md)
* [Terminal, git, and GitHub](terminal_git_github.md)
* [Cluster setup](cluster_setup.md) - specific to Longleaf cluster at UNC
* [Snakemake](snakemake.md)

# Other details

* [downloading files](download.md) from European Nucleotide Archive
* [quantify transcripts with RNA-seq](quantify.md) 
* [g2gtools](g2gtools.md) or [EMASE](emase.md) on cluster
* [aligning sequencing reads](align.md) to genome
* [ReportingTools](rtools.md) to build a differential expression analysis report
* [FASTQC and MultiQC](fastqc_multiqc.md) for quality control of sequence data
* *IGV* for exploring genomic files

